gcloud-java-dns/src/test/java/com/google/gcloud/dns/ZoneTest.java

```diff
@@ -0,0 +1,750 @@
+      .build();
+  private static final ZoneInfo NO_ID_INFO = ZoneInfo.builder(ZONE_NAME)
+      .dnsName("anoter-example.com")
+      .creationTimeMillis(893123464L)
+      .build();
+  private static final ZoneInfo NO_NAME_INFO = ZoneInfo.builder(ZONE_ID)
+      .dnsName("one-more-example.com")
+      .creationTimeMillis(875221546464L)
+      .build();
+  private static final Dns.ZoneOption ZONE_FIELD_OPTIONS =
+      Dns.ZoneOption.fields(Dns.ZoneField.CREATION_TIME);
+  private static final Dns.DnsRecordListOption DNS_RECORD_OPTIONS =
+      Dns.DnsRecordListOption.dnsName("some-dns");
+  private static final Dns.ChangeRequestOption CHANGE_REQUEST_FIELD_OPTIONS =
+      Dns.ChangeRequestOption.fields(Dns.ChangeRequestField.START_TIME);
+  private static final Dns.ChangeRequestListOption CHANGE_REQUEST_LIST_OPTIONS =
+      Dns.ChangeRequestListOption.fields(Dns.ChangeRequestField.START_TIME);
+  private static final ChangeRequest CHANGE_REQUEST = ChangeRequest.builder().id("someid").build();
+  private static final ChangeRequest CHANGE_REQUEST_AFTER = CHANGE_REQUEST.toBuilder()
+      .startTimeMillis(123465L).build();
+  private static final ChangeRequest CHANGE_REQUEST_NO_ID = ChangeRequest.builder().build();
+
+  private Dns dns;
+  private Zone zone;
+  private Zone zoneNoName;
+  private Zone zoneNoId;
+
+  @Before
+  public void setUp() throws Exception {
+    dns = createStrictMock(Dns.class);
+    zone = new Zone(dns, ZONE_INFO);
+    zoneNoId = new Zone(dns, NO_ID_INFO);
+    zoneNoName = new Zone(dns, NO_NAME_INFO);
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    verify(dns);
+  }
+
+  @Test
+  public void testConstructor() {
+    replay(dns);
+    assertNotNull(zone.info());
+    assertEquals(ZONE_INFO, zone.info());
+    assertNotNull(zone.dns());
+    assertEquals(dns, zone.dns());
+  }
+
+  @Test
+  public void testGetById() {
+    expect(dns.getZone(ZONE_ID)).andReturn(ZONE_INFO);
+    expect(dns.getZone(ZONE_ID, ZONE_FIELD_OPTIONS)).andReturn(ZONE_INFO); // for options
+    replay(dns);
+    Zone retrieved = Zone.get(dns, ZONE_ID);
+    assertSame(dns, retrieved.dns());
+    assertEquals(ZONE_INFO, retrieved.info());
+    BigInteger id = null;
+    try {
+      Zone.get(dns, id);
+      fail("Cannot get null zone.");
+    } catch (NullPointerException e) {
+      // expected
+    }
+    try {
+      Zone.get(null, id);
+      fail("Cannot get null zone.");
+    } catch (NullPointerException e) {
+      // expected
+    }
+    // test passing options
+    Zone.get(dns, ZONE_ID, ZONE_FIELD_OPTIONS);
+  }
+
+  @Test
+  public void testGetByName() {
+    expect(dns.getZone(ZONE_NAME)).andReturn(ZONE_INFO);
+    expect(dns.getZone(ZONE_ID, ZONE_FIELD_OPTIONS)).andReturn(ZONE_INFO); // for options
+    replay(dns);
+    Zone retrieved = Zone.get(dns, ZONE_NAME);
+    assertSame(dns, retrieved.dns());
+    assertEquals(ZONE_INFO, retrieved.info());
+    String name = null;
+    try {
+      Zone.get(dns, name);
```

Comment:
As of now, the method is overloaded for the second parameter being String or BigInteger. The variable is used instead of a cast on null. However, if it is the preferred way, I am switching it into `Zone.get(dns, (String) null)`;